core/iwasm/fast-jit/fe/jit_emit_control.c

```diff
@@ -917,43 +833,36 @@ jit_compile_op_br(JitCompContext *cc, uint32 br_depth, uint8 **p_frame_ip)
         SET_BB_END_BCIP(cc->cur_basic_block, *p_frame_ip - 1);
     }
 
-    return handle_next_reachable_block(cc, p_frame_ip);
+    return true;
 fail:
     return false;
 }
 
+bool
+jit_compile_op_br(JitCompContext *cc, uint32 br_depth, uint8 **p_frame_ip)
+{
+    return handle_op_br(cc, br_depth, p_frame_ip)
+           && handle_next_reachable_block(cc, p_frame_ip);
+}
+
 bool
 jit_compile_op_br_if(JitCompContext *cc, uint32 br_depth, uint8 **p_frame_ip)
 {
     JitFrame *jit_frame;
     JitBlock *block_dst;
-    JitReg frame_sp_dst, cond;
+    JitReg cond;
     JitBasicBlock *cur_basic_block, *if_basic_block = NULL;
-    JitValueSlot *frame_sp_src = NULL;
-    JitInsn *insn;
-    bool copy_arities;
-    uint32 offset;
-
-#if 0 /* TODO */
-#if WASM_ENABLE_THREAD_MGR != 0
-    /* Insert suspend check point */
-    if (cc->enable_thread_mgr) {
-        if (!check_suspend_flags(cc, func_ctx))
-            return false;
-    }
-#endif
-#endif
 
     if (!(block_dst = get_target_block(cc, br_depth))) {
         return false;
     }
 
+    /* append IF to current basic block */
     POP_I32(cond);
 
     jit_frame = cc->jit_frame;
     cur_basic_block = cc->cur_basic_block;
     gen_commit_values(jit_frame, jit_frame->lp, jit_frame->sp);
-    clear_values(jit_frame);
```

Comment:
Should not remove this line, current basic block's register may be used by other basic block, for example:
```wast
(module
  (func $test (export "test") (param $l0 i32) (result i32)
    i32.const 1
    (block $l (param i32) (result i32)
      i32.const 2
      i32.add
      local.get $l0
      local.get $l0
      br_if $l
      drop
      local.get $l0
      i32.add
    )
  )
)
```
The IRs in L3 generated are:
```bash
L3:
    ; PREDS()
    ; BEGIN_BCIP=0x002f END_BCIP=0x002f
    ADD             I21, I0, 0x94L
    STI32           i8, I21, 0x0                  => unknown/unloaded i8 used
    JMP             L4
    ; SUCCS(L4)
```
The running results:
```bash
iwasm -f test test.wasm 1
# return 3, the right result should be 1
```